### ISSUE TITLE
KaaS Staged Deployments

### ISSUE THREAD

**Issue by alex-dabija** [giantswarm/roadmap#620]
## User Stories

_- As a developer, I want to have a staged approach to deployments in order simplify development and testing of new features._

_- As a developer, I want to use the same tools in development and testing for deployments that I use for rolling out features to production in order to not use error prone hacks and workarounds._

_- As a customer, I want to be sure that during a freeze period no deployments are rolled out to specific management clusters._

## Details, Background

Currently, KaaS releases are made available to all installations as soon as the PR is merged. Testing them requires developers to manually make them available on the test management cluster(s). This situation has led to ArgoCD overwriting the manual changes. Also, it's a manual procedure which I think should be avoided.

Going forward, CAPI controllers are going to be continuously deployed to management clusters. Using the same tools in development and testing should ease the developer experience,lead to fewer

**Comment by weatherhog** [comment]
@alex-dabija this one is from 2021. Can you please check if this request is still valid?

**Comment by alex-dabija** [comment]
> @alex-dabija this one is from 2021. Can you please check if this request is still valid?

Yes, the request is still valid and it's coming from KaaS in order to have an easier time managing our clusters (management and workload).

Recently, we wanted to focus on this for management clusters with Bigmac. Those plans need to be reevaluated.

**Comment by weatherhog** [comment]
@piontec we need to decompose this issue into smaller chunks

**Comment by weatherhog** [comment]
@piontec could you please give an update on this epic ?
